src/app/content/components/Sidebar.tsx

```diff
@@ -32,12 +33,22 @@ const SidebarPlaceholder = styled.div<{isOpen: boolean}>`
   `}
 `;
 
-const SidebarControl = styled(({isOpen, ...props}: React.HTMLProps<HTMLButtonElement> & {isOpen: boolean}) =>
-  <button {...props}>
-    <span></span>
-    <span></span>
-    <span></span>
-  </button>
+type StyledSidebarControlProps = React.HTMLProps<HTMLButtonElement> & {
+  isOpen: boolean,
+};
+
+const SidebarControl = styled(({isOpen, ...props}: StyledSidebarControlProps) =>
+  <FormattedMessage id={isOpen ? 'i18n:toc:toggle:opened' : 'i18n:toc:toggle:closed'}>
+    {(txt) => {
+      return typeof txt === 'string' && <button {...props}
```

Comment:
i see that you're doing this because `aria-label` can't be an `Element` but i find this formatting to be a little gross *and* if for some reason it was an Element you'd want something more dramatic than the button silently disappearing to happen right?

i'd suggest making an `assertString` in `src/app/utils.ts` where i made an `assertDefined` helper for a similar purpose. that way you can do `aria-label={assertString(txt)}` and typescript will be ok because it either returns the thing as a string type or throws an exception.